pepper-apis/src/main/java/org/broadinstitute/ddp/event/dsmtask/impl/updateprofile/UpdateFirstLastNameHandler.java

```diff
@@ -0,0 +1,45 @@
+package org.broadinstitute.ddp.event.dsmtask.impl.updateprofile;
+
+import static org.broadinstitute.ddp.event.dsmtask.api.DsmTaskLogUtil.infoMsg;
+import static org.slf4j.LoggerFactory.getLogger;
+
+
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.TransactionWrapper.DB;
+import org.broadinstitute.ddp.db.dao.UserProfileDao;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.user.UserProfile;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+
+public class UpdateFirstLastNameHandler {
+
+    private static final Logger LOG = getLogger(UpdateFirstLastNameHandler.class);
+
+    public void doIt(String userGuid, String firstName, String lastName) {
+        TransactionWrapper.useTxn(DB.APIS, handle -> updateFirstLastName(handle, userGuid, firstName, lastName));
+    }
+
+    private void updateFirstLastName(Handle handle, String userGuid, String firstName, String lastName) {
+        var profileDao = handle.attach(UserProfileDao.class);
+        UserProfile profile = profileDao.findProfileByUserGuid(userGuid).orElse(null);
```

Comment:
I'm not sure we want to make a call out to auth0 to see if things need to be changed.  We can speed things up for the majority of cases (where values are changing) by skipping this step and just always issuing an update, even if the values are the same.

Reply:
Refactored